packages/row/src/react/element-of-type.js

```diff
@@ -0,0 +1,78 @@
+// NOTE: largely taken from `prop-types` and `react-element-proptypes`
```

Comment:
What part is custom to us?  Or, would it be possible to just use react-element-proptypes without needing to maintain this custom code?

Reply:
I opted to copy and modify because `react-element-proptypes` hasn't had any contribution for 2 years. also there are some things that looked to have changed in how react does things so I went with writing our own for now